xarray/core/variable.py

```diff
@@ -288,11 +299,26 @@ def load(self):
         because all xarray functions should either work on deferred data or
         load data automatically.
         """
-        self._data_cached()
+        new_data = self._data_cached()
```

Comment:
I would just assign unambiguously here, e.g.,
```
self._data = self._data_cast()
```
It does the same thing just a little simpler (no need to account for dask separately).